### PR TITLE
pupnp: Bump to 1.14.2, fix fpic

### PR DESCRIPTION
**Pull request by whyman**
Specify library name and version:  **pupnp/1.14.2**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/pupnp/all/conandata.yml
Comment (SpaceIm) [inline recipes/pupnp/all/conandata.yml:4]: it is not allowed to remove an existing version

File: recipes/pupnp/all/conanfile.py
Comment (SpaceIm) [inline recipes/pupnp/all/conanfile.py:43]: Could you explain why these options should be removed from the recipe?

File: recipes/pupnp/all/conandata.yml
Comment (whyman) [reply recipes/pupnp/all/conandata.yml:4]: Even if the version has CVE?

Comment (SpaceIm) [reply recipes/pupnp/all/conandata.yml:4]: question to ask to @uilianries @SSE4 and @danimtb

Comment (whyman) [reply recipes/pupnp/all/conandata.yml:4]: @uilianries @SSE4 @danimtb, any comment? I did search but there is nothing relevant that I could find.

In e.g. Linux distros packages with known vulnerability are generally removed from the repo.

Comment (danimtb) [reply recipes/pupnp/all/conandata.yml:4]: we have to discuss the policy regarding this. Thanks a lot for the ping

**Review by danimtb — CHANGES_REQUESTED** [review]
blocking the PR to prevent automatic merge by the bot

File: recipes/pupnp/all/conandata.yml
Comment (whyman) [reply recipes/pupnp/all/conandata.yml:4]: Any news @danimtb?

**Comment by whyman** [comment]
I have readded 1.12.1 to get this unblocked for now, but it will be good to hear what @danimtb and co have decided.

File: recipes/pupnp/all/conandata.yml
Comment (danimtb) [reply recipes/pupnp/all/conandata.yml:7]: Hi @whyman,
We are not removing binaries from ConanCenter binary repository and we try to keep this repo in sync for the moment, so we will not be removing the version affected from the conandata.yml. Please create a new entry for the new version. Of course, patches for declared CVEs will be welcomed!

**Comment by prince-chrismc** [comment]
This PR looks good, thanks to adding back that version...

Sadly you've been caught by new hooks, please update the recipe... you can copy paste the code from the docs if I am not mistaken, should be easy 🤞

**Comment by prince-chrismc** [comment]
@danimtb When you have a chance, please review since you are blocking it 🙏 


PR | By | Recipe | Reviews | 🛑 Blockers | 🌟 Approvers
-- | -- | -- | -- | -- | --
#5008 | whyman | 📝 pupnp | 9 | @danimtb | prince-chrismc

**Review by danimtb — APPROVED** [review]

**Review by SpaceIm — APPROVED** [review]

**Comment by whyman** [comment]
Thanks all!

**Pull request merged**
